src/shared/components/Workspace/Queries/Query/QueryComponent.tsx

```diff
@@ -0,0 +1,178 @@
+    updateList({
+      ...props.list,
+      query: {
+        ...props.list.query,
+        filters: {
+          ...props.list.query.filters,
+          ...value,
+        },
+      },
+    });
+  };
+
+  const fetchablePaginatedList: FetchableState<PaginatedList<Resource>> = {
+    error,
+    isFetching,
+    data: (data && data.resources) || null,
+  };
+  return (
+    <div className="query-component">
+      <h3 className="header">
+        <RenameableItem
+          defaultValue={`${name} ${isFetching ? 'loading' : ''}`}
+          onChange={handleUpdate}
+          size="small"
+        />
+        <div className="count">
+          {fetchablePaginatedList.data &&
+            !!fetchablePaginatedList.data.total &&
+            `${fetchablePaginatedList.data.total} result${
+              fetchablePaginatedList.data.total > 1 ? 's' : ''
+            }`}
+        </div>
+        <Icon type="close" className="close-button" onClick={handleDelete} />
+      </h3>
+      <div className="controls">
+        <FullTextSearch
+          onSearch={handleOnSearch}
+          value={query && query.textQuery}
+        />
+        <Tooltip title="Clear filters">
+          <Button icon="close-circle" onClick={handleClear} />
+        </Tooltip>
+        <Tooltip title="Refresh list">
+          <Button icon="reload" onClick={handleRefreshList} />
+        </Tooltip>
+        <Tooltip title="Clone this query">
+          <Button icon="switcher" onClick={handleCloneList} />
+        </Tooltip>
+        <Tooltip title="View ElasticSearch query">
+          <Button icon="search" onClick={() => goToQuery(props.list)} />
+        </Tooltip>
+        <TypesFilter
+          filters={query && query.filters}
+          types={data && data.types}
+          onChange={handleFilterChange}
+        />
+        <SchemasFilter
+          filters={query && query.filters}
+          schemas={data && data.schemas}
+          onChange={handleFilterChange}
+        />
+      </div>
+      <InfiniteScroll
+        makeKey={(resource: Resource) => resource.id}
+        itemComponent={(resource: Resource, index: number) => {
+          return (
+            <div key={resource.id}>
+              <ListItem
+                onClick={handleOnClick(resource)}
+                label={
+                  <Popover
+                    content={
+                      <ResourceMetadataCard
+                        {...{ ...resource, name: resource.name }}
+                      />
+                    }
+                    mouseEnterDelay={MOUSE_ENTER_DELAY}
+                    key={resource.id}
+                  >
+                    {resource.name}
+                  </Popover>
+                }
+                id={resource.id}
+                details={
+                  resource.type && resource.type.length ? (
+                    <TypesIconList type={resource.type} />
+                  ) : (
+                    undefined
```

Comment:
Bad practice to manually use `undefined`, use `null` instead or an empty JSX element `<></>`